src/Data/Text/PrettyPrinter/Markdown.hs

```diff
@@ -0,0 +1,162 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Data.Text.PrettyPrinter.Markdown (
+  SimpleMarkdown (),
+  bold,
+  italics,
+  h1,
+  h2,
+  codeBlock,
+  codeInline,
+  renderMarkdown,
+  renderAnsiToMarkdown,
+) where
+
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Data.Text.Lazy qualified as TextLazy
+import Data.Text.Lazy.Builder qualified as TextLazyBuilder
+import Prettyprinter (
+  Doc,
+  SimpleDocStream (..),
+  annotate,
+ )
+import Prettyprinter.Internal (textSpaces)
+import Prettyprinter.Render.Terminal.Internal (
+  AnsiStyle (SetAnsiStyle),
+ )
+import Prettyprinter.Render.Util.Panic (
+  panicInputNotFullyConsumed,
+  panicPeekedEmpty,
+  panicPoppedEmpty,
+  panicUncaughtFail,
+  panicUnpairedPop,
```

Comment:
[required]: These make me nervous.  I'm really not a fan of partial functions.  Let's chat to make sure these are absolutely necessary before merging.

Reply:
I can live without markdown rendering, it was neat, but is not required so removing this capability